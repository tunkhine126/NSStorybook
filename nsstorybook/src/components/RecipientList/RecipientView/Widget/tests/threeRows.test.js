/**
 *
 * Tests for ThreeRows
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import TestingComponent from 'components/shared/Testing';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import ThreeRows from '../threeRows';

describe('<ThreeRows />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <ThreeRows row1="FirstName" row2="Middle" row3="LastName" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <ThreeRows row1="FirstName" row2="Middle" row3="LastName" />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
