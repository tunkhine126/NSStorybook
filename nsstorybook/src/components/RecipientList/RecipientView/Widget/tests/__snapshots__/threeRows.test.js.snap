// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThreeRows /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start"
>
  <div
    class="MuiGrid-root makeStyles-captionSpacing-204 MuiGrid-item"
  >
    <p
      class="MuiTypography-root makeStyles-threeRowTitle-240 MuiTypography-body1"
    >
      FirstName
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <p
      class="MuiTypography-root makeStyles-contentBody-203 MuiTypography-body2"
    >
      <span
        class="styles__ThreeRowsStyle-qa8k4k-7 dyUFBs"
      >
        Middle
      </span>
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <span
      class="MuiTypography-root makeStyles-caption-201 MuiTypography-caption"
    >
      LastName
    </span>
  </div>
</div>
`;
